cli/lib/nf3.mjs

```diff
@@ -570,6 +570,25 @@ class Nf3 {
     );
   }
 
+  /**
+  Registers a proposer locally with the Optimist instance only.  This will cause
+  Optimist to make blocks when this proposer is current but these will revert if
+  the proposer isn't registered on the blockchain too.  This method is useful only
+  if the proposer is already registered on the blockchain (has paid their bond) and
+  for some reason the Optimist instance does not know about them, e.g. a new instance
+  has been created. The method 'registerProposer' will both register the proposer
+  with the blockchain and register locally with the optimist instance. So, if
+  that method has been used successfully, there is no need to also call this method
+  @method
+  @async
+  @returns {Promise} A promise that resolves to the Ethereum transaction receipt.
+  */
+  async registerProposerLocally() {
```

Comment:
It's for consistency of return.  When we call `registerProposer` we're expecting the proposer receive, sign and submit an Ethereum transaction, or fail. When we call `registerProposerLocally, we're expecting a simple success or fail report.

Reply:
I see... Can I suggest we instead return the raw transaction if it needs signing, and nothing (or a boolean or something) if it's registered locally? Then we could just make something like:

```js
// pseudo-code obviously
const { rawTx } = await nf3proposer.registerProposer()
if (rawTx) web3.submitTransaction(rawTx)
```

I really don't like that the developer needs to know the internals of optimist to understand why `registerProposer` reverts, and why there is a `registerProposerLocally`. I'd make this as abstracted as possible...

Reply:
I'm happy to do that if it makes more sense to you.  So you'd have a single endpoint and logic to process the return, i.e.:
A transaction for a first time registration
A `true` for a successful local registration
A `false` or `Error` for an attempted but un-needed re-registration.
This would be a breaking change but if it makes more sense to you like that then it probably makes more sense to everyone else.

Reply:
I can't speak for everyone else, so let's discuss this offline